website/src/parsers/js/typescript-eslint-parser.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import defaultParserInterface from './utils/defaultESTreeParserInterface';
+import pkg from 'typescript-eslint-parser/package.json';
+import SettingsRenderer from '../utils/SettingsRenderer';
+
+const ID = 'typescript-eslint-parser';
+
+const defaultOptions = {
+  range: true,
+  loc: false,
+  tokens: false,
+  comment: false,
+  tolerant: true,
+  useJSXTextNode: false,
+
+  ecmaFeatures: {
+    jsx: true,
+  },
+};
+
+const parserSettingsConfiguration = {
+  fields: [
+    'range',
+    'loc',
+    'tokens',
+    'comment',
+    'tolerant',
+    'useJSXTextNode',
+    {
+      key: 'ecmaFeatures',
+      title: 'ecmaFeatures',
+      fields: Object.keys(defaultOptions.ecmaFeatures),
+      settings:
+        settings => settings.ecmaFeatures || {...defaultOptions.ecmaFeatures},
+    },
+  ],
+};
```

Comment:
I think you need to pass `required=` with `range` option (see https://github.com/fkling/astexplorer/blob/a4edbaa49c1983ae4e0c9de2f460784f11afb099/website/src/parsers/lua/luaparse.js#L24), otherwise it will be possible to disable `range` option which will render highlighter useless (`nodeToRange` will not work at all).

Alternatively, check if `typescript-eslint-parser` returns `start` & `end` properties on the nodes (like Acorn-derived parsers do), in which case use them for `nodeToRange` instead of `.range` itself.